examples/passing/2542.purs

```diff
@@ -0,0 +1,12 @@
+module Main where
+
+import Control.Monad.Eff.Console
+
+type T = forall a. Array a
+
+foo :: T
+foo = bar where
+  bar :: forall a. Array a
+  bar = []
```

Comment:
What do you think about this being a warning test instead of a passing test?  This would have passed before the change - as our passing tests ignore warnings (if I recall correctly).